packages/lab/src/calendar/CalendarWeekHeader.tsx

```diff
@@ -1,28 +1,32 @@
 import { makePrefixer } from "@salt-ds/core";
 import { clsx } from "clsx";
 import { type ComponentPropsWithRef, forwardRef } from "react";
-import { daysForLocale } from "./internal/utils";
+import { daysOfWeek } from "./internal/utils";
 
 import { useComponentCssInjection } from "@salt-ds/styles";
 import { useWindow } from "@salt-ds/window";
+import { type DateFrameworkType, useLocalization } from "../date-adapters";
 import calendarWeekHeaderCss from "./CalendarWeekHeader.css";
-import { getCurrentLocale } from "./formatDate";
+import { useCalendarContext } from "./internal/CalendarContext";
 
-export type CalendarWeekHeaderProps = ComponentPropsWithRef<"div"> & {
-  locale?: string;
-};
+/**
+ * Props for the CalendarWeekHeader component.
+ */
+export type CalendarWeekHeaderProps = ComponentPropsWithRef<"div"> & {};
 
 const withBaseName = makePrefixer("saltCalendarWeekHeader");
 
-export const CalendarWeekHeader = forwardRef<
-  HTMLDivElement,
-  CalendarWeekHeaderProps
->(function CalendarWeekHeader(
-  { className, locale = getCurrentLocale(), ...rest },
-  ref,
-) {
-  const weekdaysShort = daysForLocale("narrow", locale);
-  const weekdaysLong = daysForLocale("long", locale);
+export const CalendarWeekHeader = forwardRef(function CalendarWeekHeader<
+  TDate extends DateFrameworkType,
+>(props: CalendarWeekHeaderProps, ref: React.Ref<HTMLDivElement>) {
+  const { className, ...rest } = props;
+  const { dateAdapter } = useLocalization<TDate>();
```

Comment:
I don't think first day of week is handled by adaptor. Also week shorthand (e.g. S/M/T), is being translated by locale.

Reply:
Confirmed working in `0.0.0-snapshot-20241106085302`